### DISCUSSION TITLE
msstore source? / Microsoft Whiteboard

### DISCUSSION THREAD

**Discussion by jaredando** [Romanitho/Winget-Install#90] [Q&A]
Is there a way to specify the source? I can't seem to pull down Microsoft.Whiteboard without this.

Weirdly enough your WingetGUI pulls it from the list of installed apps, and then lets me "reinstall" it. Not sure how to mimic it for Winget-Install. 

Your help would be really appreciated.

**Comment by Romanitho** [comment]
Hi,
This is because Winget source is forced. It was set by default due to strange behavior with Winget.

Now, we could have an argument in the script to personalize the source (keeping winget as default) that would allow you to have msstore for example (or any custom source that could be used in corporate env.)

**Reply by jaredando** [reply]
That would certainly be a feature request. 

In the meantime, could you point me in the right direction to modify your script so I could specify msstore? 
I attempted to try doing this but couldn't get it to work.

**Reply by Romanitho** [reply]
change all the `-s winget` by `-s msstore`. Not tested, but it should do the trick.

**Reply by JaredSeth** [reply]
I can confirm that this does work. I just had to use it to install the HP Command Center, since it's not available through the normal Intune MS Store integration.

**Reply by jaredando** [reply]
So I did that, and it works if I run it on the device. But when I run it through SCCM, as either User or System, it says "9MSPC6MP8FM4" does not exist. If you see the image below from the logs. The first block is run via sccm, the second i run via powershell manually (navigating to the ccmcache to ensure continuity.) 

![image](https://github.com/Romanitho/Winget-Install/assets/148944176/2161693c-47ab-4ebe-b29c-3c9c317c4855)

Unless there is a better may to reference microsoft whiteboard? 
I love this framework but for Whiteboard its been very tricky.

**Reply by KnifMelti** [reply]
SYSTEM can't pull from the Store, only users

**Reply by jaredando** [reply]
This resolved it! Legend, maybe I didn't try both system and user like I thought
